### PR TITLE
Feature/zcs 11179

### PR DESCRIPTION
**Pull request by SuryawanshiAmol**

### CODE REVIEW

**Comment by ghost** [comment]
When I try to compile NGinx for Ubuntu 20, it works, but there is an error when creating the source package. This means the binary package works, but the source package is not created. I am not sure if we need to provide the source package (to comply with the license) since we already release our patch files. So perhaps this step can be removed.

```
make[1]: Leaving directory '/tmp/packages/thirdparty/nginx/build/UBUNTU20_64/zimbra-nginx'
 dpkg-source -irpm -b .
dpkg-source: info: using source format '3.0 (quilt)'
dpkg-source: info: building zimbra-nginx using existing ./zimbra-nginx_1.24.0.orig.tar.gz
dpkg-source: info: using patch list from debian/patches/series
dpkg-source: error: cannot represent change to objs/addon/nviennot-nginx-tcp-keepalive/ngx_http_tcp_keepalive_module.o: binary file contents changed
dpkg-source: error: add objs/addon/nviennot-nginx-tcp-keepalive/ngx_http_tcp_keepalive_module.o in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: error: cannot represent change to objs/addon/sso/ngx_http_zm_sso_module.o: binary file contents changed
dpkg-source: error: add objs/addon/sso/ngx_http_zm_sso_module.o in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: error: cannot represent change to objs/addon/throttle/ngx_mail_throttle_module.o: binary file contents changed
dpkg-source: error: add objs/addon/throttle/ngx_mail_throttle_module.o in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: error: cannot represent change to objs/addon/upstreamzmauth/ngx_http_upstream_zmauth_module.o: binary file contents changed
dpkg-source: error: add objs/addon/upstreamzmauth/ngx_http_upstream_zmauth_module.o in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: error: cannot represent change to objs/addon/zmauth/ngx_mail_zmauth_module.o: binary file contents changed
dpkg-source: error: add objs/addon/zmauth/ngx_mail_zmauth_module.o in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: error: cannot represent change to objs/nginx: binary file contents changed
dpkg-source: error: add objs/nginx in debian/source/include-binaries if you want to store the modified binary in the debian tarball
dpkg-source: warning: executable mode 0755 of 'objs/nginx' will not be represented in diff
...
repeats 100 or so more times
```

**Comment by barrydegraaff** [comment]
Please explain how to build using this change.

File: thirdparty/nginx/patches/nginx_README.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+--- nginx/README.md	1970-01-01 05:30:00.000000000 +0530
++++ nginx/README.md	2023-09-14 18:47:11.380437400 +0530
+@@ -0,0 +1,11 @@
++# nginx
++
++This is the official repository for building out the third party dependency nginx for Zimbra Collaboration Suite 8.7 and later.
++
++Issues should be reported via [Zimbra's bugzilla](https://bugzilla.zimbra.com)
```
Comment (silentsakky) [inline thirdparty/nginx/patches/nginx_README.patch]: bugzilla is no more :)

**Review by ghost — APPROVED** [review]
It is a little difficult to verify the patches from https://github.com/Zimbra/packages/pull/194 are the same as what is in https://github.com/Zimbra/nginx/pull/5 because we do not apply our changes on the upstream nginx source. Instead we copy upstream and apply them along with all our other nginx modules and then also remove and move files.

After manually moving and removing some files,  I can confirm that this https://github.com/Zimbra/packages/pull/194 produces the same as https://github.com/Zimbra/nginx/pull/5

The source package throws many errors for me such as 
```
dpkg-source: error: cannot represent change to objs/addon/nviennot-nginx-tcp-keepalive/ngx_http_tcp_keepalive_module.o: binary file contents changed
```

But since that source package is AFAIK not used, the PR is _good enough_.

Verified using:

```
#!/bin/bash

cd /tmp
rm packages zimbra-* -Rf

apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 9BE6ED79

rm /etc/apt/sources.list.d/zimbra.list
cat >> /etc/apt/sources.list.d/zimbra.list << EOF
deb [arch=amd64] https://repo.zimbra.com/apt/87 focal zimbra
deb-src [arch=amd64] https://repo.zimbra.com/apt/87 focal zimbra
deb [arch=amd64] https://repo.zimbra.com/apt/1000 focal zimbra
deb-src [arch=amd64] https://repo.zimbra.com/apt/1000 focal zimbra
deb [arch=amd64] https://repo.zimbra.com/apt/1000-ne focal zimbra
deb-src [arch=amd64] https://repo.zimbra.com/apt/1000-ne focal zimbra
EOF

apt-get clean
apt update

apt-get source gkrellweather
apt-get build-dep gkrellweather

apt -y install m4 libpcre3-dev build-essential git binutils lintian debhelper dh-make devscripts libz-dev quilt

GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" git clone git@github.com:Zimbra/packages.git
GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" git clone git@github.com:Zimbra/zimbra-build.git
GIT_SSH_COMMAND="ssh -o UserKnownHostsFile=/dev/null -o StrictHostKeyChecking=no" git clone git@github.com:Zimbra/zimbra-package-stub.git

cd zimbra-package-stub
git checkout develop #ubuntu 20 support
cd ..

cd packages
git checkout feature/ZCS-11179
git submodule update --init --recursive --remote

cd /tmp/packages/thirdparty/nginx/
cd nginx
#git checkout feature/ZCS-11179
rm -Rf nginx
cd ..
make

sleep 10

make build

echo "Compiled version:"
/tmp/packages/thirdparty/nginx/build/UBUNTU20_64/zimbra-nginx/objs/nginx -V

```

File: thirdparty/nginx/patches/nginx_README.patch (hunk 1 above)
Comment (SuryawanshiAmol) [reply thirdparty/nginx/patches/nginx_README.patch]: Yes. Will add new Jira link here.

**Pull request merged**